controllers/agents/app/serviceclaim_controller.go

```diff
@@ -43,11 +61,141 @@ type ServiceClaimReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *ServiceClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	l := log.FromContext(ctx).WithValues("namespace", req.Namespace, "name", req.Name)
+	l.Info("Reconciling service claim")
+
+	var sclaim primazaiov1alpha1.ServiceClaim
+	if err := r.Get(ctx, req.NamespacedName, &sclaim); err != nil {
```

Comment:
Let's use the `apierrors k8s.io/apimachinery/pkg/api/errors` IsNotFound , instead of checking if error is nil

Reply:
Done